array.go

```diff
@@ -1193,3 +1193,143 @@ func countElement(array []interface{}, element interface{}) int {
+// For default, it will use built-in comparator function for each data type.
+// It requires all elements in the array to have same data type
+//
+// Example:
+//
+//	array := NewArray(t, []interface{}{100, 101, 102})
+//	array.IsOrdered(func(x, y *Value) bool {
+//		valX, _ := x.Raw().(int)
+//		valY, _ := y.Raw().(int)
+//		return valX >= valY
+//	})
+func (a *Array) IsOrdered(less ...func(x, y *Value) bool) *Array {
+	a.chain.enter("IsOrdered()")
+	defer a.chain.leave()
+
+	if a.chain.failed() {
+		return a
+	}
+
+	if len(less) > 1 {
+		a.chain.fail(AssertionFailure{
+			Type: AssertUsage,
+			Errors: []error{
+				errors.New("unexpected multiple less arguments"),
+			},
+		})
+		return a
+	}
+
+	if len(a.value) == 0 {
+		return a
+	}
+
+	var fn func(x, y *Value) bool
+	if len(less) == 1 {
+		fn = less[0]
+	} else {
+		// use default comparator based on data type
+		cArr, ok := canonArray(a.chain, a.value)
+		if !ok {
+			return a
+		}
+
+		fn = a.getDefaultComparator(cArr)
+		if a.chain.failed() {
+			return a
+		}
+	}
+
+	for i := 0; i < int(a.Length().Raw())-1; i++ {
+		xChain := a.chain.clone()
+		xChain.replace("IsOrdered[%d]", i)
+		x := newValue(xChain, a.value[i])
+
+		yChain := a.chain.clone()
+		yChain.replace("IsOrdered[%d]", i+1)
+		y := newValue(yChain, a.value[i+1])
+
+		if !fn(x, y) {
+			a.chain.fail(AssertionFailure{
+				Type:      AssertLt,
+				Actual:    &AssertionValue{x.value},
+				Expected:  &AssertionValue{y.value},
+				Reference: &AssertionValue{a.value},
+				Errors: []error{
+					errors.New("expected: array is ordered"),
+					fmt.Errorf("element %v must be less than element %v, but it isn't",
+						x.value, y.value),
+				},
+			})
+			return a
+		}
+	}
+
+	return a
+}
+
+func (a *Array) getDefaultComparator(cArr []interface{}) func(x, y *Value) bool {
+	var fn func(x, y *Value) bool
+	var currType, prevType string
+	var prevElem interface{}
+	for _, v := range cArr {
+		switch t := v.(type) {
+		case bool:
+			currType = fmt.Sprintf("%T", t)
+			fn = func(x, y *Value) bool {
+				xVal, _ := x.Raw().(bool)
+				yVal, _ := y.Raw().(bool)
+				return (!xVal && yVal) || (xVal == yVal)
+			}
+		case float64:
+			currType = fmt.Sprintf("%T", t)
+			fn = func(x, y *Value) bool {
+				xVal, _ := x.Raw().(float64)
+				yVal, _ := y.Raw().(float64)
+				return xVal <= yVal
+			}
+		case string:
+			currType = fmt.Sprintf("%T", t)
+			fn = func(x, y *Value) bool {
+				xVal, _ := x.Raw().(string)
+				yVal, _ := y.Raw().(string)
+				return xVal <= yVal
+			}
+		default:
+			currType = fmt.Sprintf("%T", t)
+			a.chain.fail(AssertionFailure{
+				Type:      AssertBelongs,
+				Actual:    &AssertionValue{currType},
+				Expected:  &AssertionValue{AssertionList{"boolean", "number", "string"}},
+				Reference: &AssertionValue{v},
+				Errors: []error{
+					errors.New(`expected: type of each element of reference array
+					 belongs to given list`),
```

Comment:
Such formatting will insert unwanted newline and spaces to the error string. Instead, use something like this:

```
			Errors: []error{
				errors.New("expected: type of each element of reference array"+
				   "belongs to given list"),
```